src/main/java/duke/Duke.java

```diff
@@ -42,70 +37,88 @@ public Duke(String filePath) {
      * delete, list. Other commands are ignored.
      */
     public void run() {
+        storage.loadTasksFromFile(tasks);
         String command;
-        boolean stillRunning = true;
+        isRunning = true;
         Scanner scanner = new Scanner(System.in);
-        while (stillRunning) {
+        while (isRunning) {
             command = scanner.nextLine();
             parser.interpretCommand(command);
             String firstCommand = this.parser.getFirstCommand();
             try {
-                switch (firstCommand) {
-                case "bye":
-                    this.ui.goodBye();
-                    stillRunning = false;
-                    break;
-                case "done":
-                    markDone();
-                    break;
-                case "deadline":
-                case "todo":
-                case "event":
-                    addTask();
-                    break;
-                case "delete":
-                    deleteTask();
-                    break;
-                case "list":
-                    this.tasks.listTasks();
-                    break;
-                case "find":
-                    String keyword = this.parser.findKeyword();
-                    ArrayList<Task> tasksWithKeyword = this.tasks.findTasksUsingKeyword(keyword);
-                    this.ui.showTasksWithKeyword(tasksWithKeyword);
-                    break;
-                default:
-                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
-                }
+                ui.respondToUser(respondToFirstCommand(firstCommand));
             } catch (DukeException e) {
                 this.ui.showError(e);
             }
         }
     }
 
+    /**
+     * Runs Duke when launched from the GUI
+     * @param command The command for Duke to handle.
+     * @return Response from Duke.
+     */
+    public String runGui(String command) {
+        try {
+            storage.loadTasksFromFile(tasks);
+            parser.interpretCommand(command);
+            String firstCommand = parser.getFirstCommand();
+            return respondToFirstCommand(firstCommand);
+        } catch (DukeException e) {
+            return e.getMessage();
+        }
+    }
+
+    private String respondToFirstCommand(String firstCommand) {
+        String response;
+        switch (firstCommand) {
+        case "bye":
```

Comment:
suggest making firstCommand ignore case or to set to lowercase to reduce user input error?